cpu/cortexm_common/thread_arch.c

```diff
@@ -265,15 +265,18 @@ void *thread_isr_stack_start(void)
     return (void *)&_sstack;
 }
 
-__attribute__((naked)) void NORETURN cpu_switch_context_exit(void)
+void NORETURN cpu_switch_context_exit(void)
 {
+    /* enable IRQs to make the SVC interrupt is reachable */
+    irq_enable();
+    /* trigger the SVC interrupt */
     __asm__ volatile (
-    "bl     irq_enable               \n" /* enable IRQs to make the SVC
```

Comment:
I don't fully understand why this apparently works in master. It stopped working when compiling with LTO.

Reply:
You mean that if there is no symbol this shouldn't work?

Reply:
Yes, indeed. 

But: Apparently there is still a symbol generated:

```
$ arm-none-eabi-readelf -s examples/default/bin/nucleo-f767zi/default.elf  | grep irq_enable
    81: 080003f1     8 FUNC    LOCAL  DEFAULT    1 irq_enable
```

So apparently if an inline function is called from assembly via its symbol name, GCC still emits the function once as full blown function and generates the required symbol. I would have preferred if GCC would not do such magic things and just get a linker error, as we that way would have noticed this while switching to the `inline`-able API.